File: 02- Estructuras de control/java/ejercicios/NumerosPrimosDentroDeRango.java
/**
 * Descripción: Solicita dos números enteros y muestra todos los números primos que se
 *      encuentren dentro de ese rango.
 * Autor: Santiago Nicolás De la mora Núñez
 * Fecha: 11/02/2026
 */

package mx.edu.uttt.ejerciciosunidad2;

import javax.swing.JOptionPane;

public class NumerosPrimosDentroDeRango {
    public static void main(String[] args) {
        // Declaración e inicialización de variables y constantes
        final int NUMERO_PRIMO_MENOR = 2;
        int numeroInicial = 0, numeroFinal = 0, numeroDivisiones = 0;
        String numerosPrimos = "";
        boolean esValido = false;

        // Solicita el número inicial
        do {
            numeroInicial = Integer.parseInt(JOptionPane.showInputDialog(null, "Ingresa número inicial"));

            // Verifica si el número es negativo, cero o uno
            if (numeroInicial < 2) {
                JOptionPane.showMessageDialog(null, "No puede ingresar un número menor a 2.");

                // Indica que se debe volver a solicitar el número
                esValido = false;
            } else {
                // Indica que se debe dejar de repetir el bucle
                esValido = true;
            }
        } while (!esValido);

        // Solicita el número final
        do {
            numeroFinal = Integer.parseInt(JOptionPane.showInputDialog(null, "Ingrese el número final"));

            // Verifica si el número es menor o igual al número inicial
            if (numeroFinal <= numeroInicial) {
                JOptionPane.showMessageDialog(null, "El número final no puede ser menor que el número inicial.");

                // Indica que se debe volver a solicitar el número
                esValido = false;
            } else {
                // Indica que se debe dejar de repetir el bucle
                esValido = true;
            }
        } while (!esValido);

        // Muestra todos los números primos dentro del rango
        for (int i = numeroInicial; i <= numeroFinal; i++) {
            // Reinicia el contador de número de divisiones
            numeroDivisiones = 0;

            for (int t = NUMERO_PRIMO_MENOR; t <= i; t++) {
                // Verifica si el número final dividido entre cada número menor a él no tiene residuo (es una división exacta)
                if (i % t == 0) {
                    // Cuenta cada división exacta
                    numeroDivisiones++;
                }
            }

            // Verifica si el número se dividió solo una vez
            if (numeroDivisiones == 1) {
                // Concatena el número y agrega un espacio
                numerosPrimos += i + "  ";
            }
        }

        // Muestra los números primos
        JOptionPane.showMessageDialog(
                null,
                "Números primos entre " + numeroInicial + " y " + numeroFinal + ":\n" +
                        numerosPrimos
        );
    }
}

File: 02- Estructuras de control/python/numeros_primos_dentro_de_rango.py
"""
Descripción: Solicita dos números enteros y muestra todos los números primos que se
    encuentren dentro de ese rango.
Autor: Santiago Nicolás De la mora Núñez
Fecha: 11/02/2026
"""

# Declaración e inicialización de variables y constantes
NUMERO_PRIMO_MENOR = 2
numero_inicial = 0
numero_final = 0
numero_divisiones = 0
numeros_primos = ""
es_valido = False

# Solicita el número inicial
while True:
    numero_inicial = int(input("Ingresa el número inicial:\n"))

    # Verifica si el número es negativo, cero o uno
    if numero_inicial < 2:
        print("\nNo puede ingresar un número menor a 2.\n")

        # Indica que se debe volver a solicitar el número
        es_valido = False
    else:
        # Indica que se debe dejar de solicitar el número
        es_valido = True

    # Rompe el bucle si el número ingresado fue válido
    if es_valido:
        break

# Solicita el número final
while True:
    numero_final = int(input("\nIngresa el número final:\n"))

    # Verifica si el número es negativo, cero o uno
    if numero_final <= numero_inicial:
        print("\nEl número final no puede ser menor que el número inicial.\n")

        # Indica que se debe volver a solicitar el número
        es_valido = False
    else:
        # Indica que se debe dejar de solicitar el número
        es_valido = True

    # Rompe el bucle si el número ingresado fue válido
    if es_valido:
        break

# Muestra todos los números primos dentro del rango
i = numero_inicial
while (i <= numero_final):
    # Reinicia el contador del número de divisiones
    numero_divisiones = 0

    t = NUMERO_PRIMO_MENOR
    while (t <= i):
        # Verifica si el número final dividido entre cada número menor a él no tiene residuo (es una división exacta)
        if i % t == 0:
            # Cuenta cada división exacta
            numero_divisiones += 1

        t += 1

    # Verifica si el número se dividió solo una vez
    if numero_divisiones == 1:
        # Concatena el número y agrega un espacio
        numeros_primos += str(i) + " "

    i += 1

# Muestra los números primos
print(f"\nNúmeros primos entre {numero_inicial} y {numero_final}:\n{numeros_primos}")